crates/chain/src/tx_graph.rs

```diff
@@ -107,19 +108,19 @@ use core::{
 ///
 /// [module-level documentation]: crate::tx_graph
 #[derive(Clone, Debug, PartialEq)]
-pub struct TxGraph<A = ()> {
+pub struct TxGraph<AM = BlockTime> {
     // all transactions that the graph is aware of in format: `(tx_node, tx_anchors)`
-    txs: HashMap<Txid, (TxNodeInternal, BTreeSet<A>)>,
+    txs: HashMap<Txid, (TxNodeInternal, BTreeMap<Anchor, AM>)>,
```

Comment:
```suggestion
    txs: HashMap<Txid, (TxNodeInternal, BTreeSet<BlockId>)>,
```
Anchor contains the txid. It does not make as part of the value of something keyed by txid.

Also we can lookup the `A` (anchor meta) from `anchors`.